x/pairing/keeper/grpc_query_user_entry.go

```diff
@@ -38,15 +39,19 @@ func (k Keeper) UserEntry(goCtx context.Context, req *types.QueryUserEntryReques
 		}
 
 		planPolicy := plan.GetPlanPolicy()
-
+		policies := []*projectstypes.Policy{&planPolicy, project.AdminPolicy, project.SubscriptionPolicy}
 		// geolocation is a bitmap. common denominator can be calculated with logical AND
-		geolocation := project.AdminPolicy.GeolocationProfile & project.SubscriptionPolicy.GeolocationProfile & planPolicy.GeolocationProfile
+		geolocation := k.CalculateEffectiveGeolocationFromPolicies(policies)
+
+		sub, found := k.subscriptionKeeper.GetSubscription(ctx, project.GetSubscription())
+		if !found {
+			return nil, fmt.Errorf("could not find subscription with address %s", project.GetSubscription())
+		}
+		allowedCU := k.CalculateEffectiveAllowedCuPerEpochFromPolicies(policies, project.GetUsedCu(), sub.GetMonthCuLeft())
 
-		allowedCU := commontypes.FindMin([]uint64{
-			project.AdminPolicy.GetEpochCuLimit(),
-			project.SubscriptionPolicy.GetEpochCuLimit(),
-			planPolicy.GetEpochCuLimit(),
-		})
+		if !projectstypes.VerifyTotalCuUsage(policies, project.GetUsedCu()) {
```

Comment:
we dont want an error here, if the user have not more cu left return allowedCU = 0